src/actions/guides/json_and_apis/parsing-json-requests.cr

```diff
@@ -7,12 +7,72 @@ class Guides::JsonAndApis::ParsingJsonRequests < GuideAction
 
   def markdown : String
     <<-MD
-    ## Processing JSON that isn’t saved to the database
+    ## Converting JSON params to objects
 
-    You can use
-    [`JSON.mapping`](https://crystal-lang.org/api/JSON.html#mapping%28_properties_%2Cstrict%3Dfalse%29-macro)
-    and [`JSON.parse`](https://crystal-lang.org/api/JSON.html) to parse
-    the response body in any way you’d like.
+    Crystal comes with a [JSON::Serializable](https://crystal-lang.org/api/1.0.0/JSON/Serializable.html) module that makes converting raw JSON in to
+    composable objects much easier. This makes validating your JSON, and storing the data nicer and type-safe.
+
+    We will use this JSON object for an example:
+
+    ```json
+    {
+      "pay_date": "2021-02-15 10:00:00",
+      "issue_date": "2021-01-31 14:00:00",
+      "lines": [
+        {"text": "Item One", "unit": "USD", "amount": 5500, "tax": 3.8},
+        {"text": "Item Two", "unit": "USD", "amount": 1032, "tax": 3.8},
+      ]
+    }
+    ```
+
+    ### Using params
+
+    The `params` object gives us access to this JSON with `params.body`. We can also access individual values with `params.from_json["the_key"]`.
+
+    ```crystal
+    class Api::Invoices::Create < ApiAction
+      post "/api/customers/:customer_id/invoices" do
+        pay_date = Time.parse(params.from_json["pay_date"].as_s, "%Y-%m-%d %H:%M:%S", Time::Location::UTC)
+
+        SaveInvoice.create(pay_date: pay_date) do |op, invoice|
+          # ...
+        end
+      end
+    end
+    ```
+
+    To simplify the code a bit, we can create separate JSON serializable objects.
+
+    ```crystal
+    class InvoiceSerializer
```

Comment:
Can this and the other be changed to `SerializedInvoice` and `SerializedInvoiceLine` to avoid confusion with Lucky's serializer classes

Reply:
yup. Good call